common/Views/AdaptiveCardViews/ContentDialogWithNonInteractiveContent.xaml.cs

```diff
@@ -15,27 +17,57 @@ namespace DevHome.Common.Views.AdaptiveCardViews;
 /// </summary>
 public sealed partial class ContentDialogWithNonInteractiveContent : ContentDialog
 {
+    private readonly IThemeSelectorService _themeSelector;
+
+    private readonly DevHomeContentDialogContent _content;
+
     public ContentDialogWithNonInteractiveContent(DevHomeContentDialogContent content)
     {
         this.InitializeComponent();
 
+        _themeSelector = Application.Current.GetService<IThemeSelectorService>();
+
         // Since we use the renderer service to allow the card to receive theming updates, we need to ensure the UI thread is used.
         var dispatcherQueue = Application.Current.GetService<DispatcherQueue>();
         dispatcherQueue.TryEnqueue(async () =>
         {
             Title = content.Title;
             PrimaryButtonText = content.PrimaryButtonText;
-            var rendererService = Application.Current.GetService<AdaptiveCardRenderingService>();
-            var renderer = await rendererService.GetRendererAsync();
-            renderer.HostConfig.ContainerStyles.Default.BackgroundColor = Microsoft.UI.Colors.Transparent;
-            var card = renderer.RenderAdaptiveCardFromJsonString(content.ContentDialogInternalAdaptiveCardJson?.Stringify() ?? string.Empty);
+            var cardContent = await MakeCardContentAsync();
             Content = new ScrollViewer
             {
                 VerticalScrollBarVisibility = ScrollBarVisibility.Auto,
-                Content = card.FrameworkElement,
+                Content = cardContent,
             };
+
+            // Set the theme of the content dialog box
+            RequestedTheme = _themeSelector.IsDarkTheme() ? ElementTheme.Dark : ElementTheme.Light;
             SecondaryButtonText = content.SecondaryButtonText;
             this.Focus(FocusState.Programmatic);
         });
+
+        _themeSelector.ThemeChanged += OnThemeChanged;
+        _content = content;
+    }
+
+    private async void OnThemeChanged(object? sender, ElementTheme newRequestedTheme)
+    {
+        // set the theme of the content dialog box.
+        RequestedTheme = newRequestedTheme;
+        var cardContent = await MakeCardContentAsync();
+        Content = new ScrollViewer
```

Comment:
Would it be easier for MakeCardContent() to return this ScrollViewer, so you don't have to remake it in a couple places?

Reply:
Technically, yes.  De-duping is always good.  Done.